pkgs/tools/misc/swaglyrics/default.nix

```
@@ -0,0 +1,31 @@
+{ stdenv, python3 }:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "swaglyrics";
+  version = "1.2.2";
+
+  src = python3.pkgs.fetchPypi {
+    inherit pname version;
+    sha256 = "1cayhzd3va3gg5qxjmbm9fc501ziz4qykcl8b0mcm0l5adpy8d05";
+  };
+
+  propagatedBuildInputs = with python3.pkgs; [
+    unidecode colorama beautifulsoup4 flask requests swspotify
+  ];
+
+  preConfigure = ''
+    substituteInPlace setup.py \
+      --replace 'requests>=2.24.0' 'requests~=2.23'
+  '';
+
+  preBuild = "export HOME=$NIX_BUILD_TOP";
+
+  checkInputs = [ python3.pkgs.pytest ];
```

to automatically execute it
```suggestion
  checkInputs = [ python3.pkgs.pytestCheckHook ];
```